widgets/splitbutton/org.eclipse.nebula.widgets.splitbutton/src/org/eclipse/nebula/widgets/splitbutton/SplitButton.java

```diff
@@ -179,30 +179,29 @@ private void drawTriangle() {
 	}
 
 	private void addMouseDownListener() {
-		super.addListener(SWT.MouseDown, new Listener() {
-
-			@Override
-			public void handleEvent(final Event event) {
-				if (shouldShowMenu(event.x, event.y)) {
-					final Button button = (Button) event.widget;
-					final Rectangle rect = button.getBounds();
-					final Point p = button.toDisplay(rect.x, rect.y + rect.height);
-					getMenu().setLocation(p.x - rect.x, p.y - rect.y);
-					getMenu().setVisible(true);
-
-				} else {
-					for (final Listener listener : listeners) {
-						final Event evt = new Event();
-						evt.widget = SplitButton.this;
-						evt.display = getDisplay();
-						evt.type = SWT.Selection;
-						listener.handleEvent(evt);
-					}
+		super.addListener(SWT.MouseDown, event -> {
+			boolean shouldShowMenu = event.x >= splitButtonAreaLeft && event.y >= splitButtonAreaTop
+					&& event.x <= splitButtonAreaRight && event.y <= splitButtonAreaBottom;
+			if (shouldShowMenu) {
+				final Button button = (Button) event.widget;
+				final Rectangle rect = button.getBounds();
+				final Point p = button.toDisplay(rect.x, rect.y + rect.height);
+				getMenu().setLocation(p.x - rect.x, p.y - rect.y);
+				getMenu().setVisible(true);
+			} else {
+				Supplier<Event> eventFactory = () -> {
+					Event evt = new Event();
+					evt.widget = SplitButton.this;
+					evt.display = getDisplay();
+					evt.type = SWT.Selection;
+					return evt;
+				};
+				for (Listener listener : selectionListeners) {
+					listener.handleEvent(eventFactory.get());

```

Comment:
Actually it looks wrong to create a new event for each listener, @lcaron what do you think?

Reply:
It used to be like that, this is why I replicated it.
But when looking at SWT I get the impression that there is only one Event send to all listeners, unless one uses a typed listener (e.g. SelectionListener), then the one event is wrapped in a new `SelectionEvent` for each `SelectionListener`, see `TypedListener`. But the latter is probably a technical detail.
And although Events are not immutable, I think good listeners don't change events.

Reply:
Except for "doit" when the event has that sort of thing.

Reply:
> Except for "doit" when the event has that sort of thing.

Didn't knew that, thanks for the hint. So maybe there should be one `Event` send to all `Listeners` and a new `SelectionEvent` for each `SelectionListener`?

Reply:
I think that's how SWT generally does it...

Reply:
> And although Events are not immutable, I think good listeners don't change events.

Actually listeners can give some things back to the caller (e.g. as mentioned doit), there is even a field `data` mentioned for "application use" in TypedEvent, I think if one can install listeners to an SWT object one should always be "nice" and only ever modify things if the API tells so.